s3transfer/tasks.py

```diff
@@ -253,7 +253,7 @@ def _main(self, transfer_future, **kwargs):
             # Call the submit method to start submitting tasks to execute the
             # transfer.
             self._submit(transfer_future=transfer_future, **kwargs)
-        except Exception as e:
+        except BaseException as e:
```

Comment:
It is changed specifically for the serial implementation. The ``SubmissionTask`` acts as the outer most layer of the main thread when the serial implementation is used and so if a ``KeyboardInterrupt`` is received during the transfer process, done will not properly get announced, causing the process to hang if using the context manager of the TransferManager, because ``Exception`` does not catch the ``KeyboardInterrupt``.

I will add a comment to make that more clear.